riscv/src/result.rs

```diff
@@ -0,0 +1,44 @@
+use core::fmt;
+
+/// Convenience alias for the [Result](core::result::Result) type for the library.
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Represents error variants for the library.
+#[repr(C)]
```

Comment:
Just curious: how does `#[repr(C)]` work with complex enums?

Reply:
Somewhat of a habit to use `#[repr(C)]` on types that could potentially cross an FFI boundary, though I'm not exactly sure how Rust handles it for complex enums.

From [godbolt](https://godbolt.org/z/hnW7q8xj7) it looks like it compiles to a flat structure with pointer offsets.

I can remove it if you want.

Reply:
I'm not an expert in these issues, but I would say that results are quite rustacean, and error types are not usually exposed in FFIs. Instead, we would expose the "infallible" functions.